### ISSUE TITLE
Doc no. of shards proportional to validator's balance

### ISSUE THREAD

**Issue by jamesray1** [ethereum/sharding#22]
The sample function should be coded in such a way that any given validator randomly gets allocated to some number of shards every `SHUFFLING_CYCLE_LENGTH`, where the expected number of shards is proportional to the validator's balance. 

My understanding is that each shard can have a differently sized coinbase. Because of this, I want to find how much Ether should be allocated to a validator in proportion to the ratio of the size of their deposit to the sum of all validator deposits. 

ETH allocated per validator / total ETH = deposit per validator / sum of all validator deposits

ETH allocated per validator = total ETH * deposit per validator / sum of all validator deposits

However, my understanding is that a validator cannot validate a portion of a shard—they can only validate a whole shard, or not validate it at all. Thus, this exact amount of ETH calculated may not be able to be assigned to any validator, because there may not be any number of shards with a sum total coinbase equal to this amount. Therefore, there would need to be some overlap (and in fact you would want more than one validator per shard, maybe with some minimum limit of validators per shard). 

So maybe instead, the number of shards per validator should at least equal to the ETH allocated per validator, while maybe each shard should have a minimum limit of validators per shard. Or alternatively, I understand that there is a minimum coinbase per shard, which can be set so that each shard doesn't have too few validators.

In summary,

n_shards / validator >= ETH allocated per validator = total ETH * deposit per validator / sum of all validator deposits

**Comment by jamesray1** [comment]
Ah I read after that: "This picks out 100 validators for each shard during each cycle".

**Comment by mhchia** [comment]
Sorry for the late reply.

> The sample function should be coded in such a way that any given validator randomly gets allocated to some number of shards every SHUFFLING_CYCLE_LENGTH, where the expected number of shards is proportional to the validator's balance.

Yep you're right

> My understanding is that each shard can have a differently sized coinbase. Because of this, I want to find how much Ether should be allocated to a validator in proportion to the ratio of the size of their deposit to the sum of all validator deposits. It should be fixed in the future I think.

In the spec I think we didn't define a differently sized coinbase for each shard. The chance being selected as a collator, i.e. selected by the `sample` function in validator manager contract should be proportional to the deposit size of the validator. However, for now we just restrict the deposit size to be 100 ETH, to make the implementation easier.

**Comment by jamesray1** [comment]
OK, thanks for clarifying.

**Comment by hwwhww** [comment]
Close all old stuff
